src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -348,6 +357,14 @@ public Snapshot retrieveSnapshotByName(String name) {
     return snapshot;
   }
 
+  private SnapshotRequestContentsModel stringToSnapshotRequestContentsModel(String json) {
+    try {
+      return objectMapper.readValue(json, SnapshotRequestContentsModel.class);
+    } catch (JsonProcessingException e) {
+      return null;
```

Comment:
Shouldn't we throw an `CorruptMetadataException` here? If the default value in the DB is an empty object, there should never be invalid JSON.

Reply:
Additionally, why not inline this code in the `try` block where its called? Then, you could catch the `JsonProcessingException` there in the same `try` block. This code doesn't appear to be used anywhere else, so I'm not sure putting it into its own method buys anything.

Reply:
That was actually my suggestion: it's going to be helpful to have this pulled out when we start having different versions of the creation specs and need to handle backwards compatibility

Reply:
you can't actually catch the JsonProcessingException in the same try block, since exists inside an anonymous function, the try - catch block must also exist inside the anonymous function

Reply:
if we throw CorruptMetadataException, we'll have to catch it in the anonymous function and return null anyways.

Reply:
The anonymous function isn't the issue, the issue is that this argument to `NamedParameterJdbcTemplate.queryForObject()` is a `RowMapper` and the only exception it can throw is `SQLException`. If the thrown exception was `SQLException` then it would compile fine.

What is the desired behavior if `objectMapper.readValue()` fails? If you want the operation to fail, I would rethrow here `SQLException`.

```
    } catch (JsonProcessingException e) {
      throw new SQLException(e);
    }
```

Reply:
And if this was Scala, we'd be trying to figure out how to convert an cats effect IO to a Future...

Reply:
I don't think we need to fail the entire operation if this field fails to parse (which should never happen, since it needs to be valid jsonb going in)

Reply:
Right...so it can be valid JSON but then if the model changes over time, it'll fail trying to read it back out (unless we come up with a migration strategy)

Reply:
I can add a logging message here for future debugging
      logger.error("Error parsing creationInfo into SnapshotRequestContentsModel", e);
and then continue on by returning null?

Reply:
I'd be good with that

Reply:
maybe log.warning